src/back/glsl/mod.rs

```diff
@@ -3499,7 +3529,7 @@ impl<'a, W: Write> Writer<'a, W> {
     fn write_zero_init_value(&mut self, ty: Handle<crate::Type>) -> BackendResult {
         let inner = &self.module.types[ty].inner;
         match *inner {
-            TypeInner::Scalar { kind, .. } => {
+            TypeInner::Scalar { kind, .. } | TypeInner::Atomic { kind, .. } => {
```

Comment:
GLSL has no concept of atomics at the type level and should be treated as scalars.
This was needed because workgroup variables might contain atomics which should be initialized just as scalars.